packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -222,6 +223,13 @@ export class CoreKibanaRequest<
       options,
     };
   }
+  /** infer route access from path if not declared */
+  private getAccess(request: RawRequest): 'internal' | 'public' {
+    return (
+      ((request.route?.settings as RouteOptions)?.app as KibanaRouteOptions)?.access ??
+      (request.path.includes('internal') ? 'internal' : 'public')
```

Comment:
This algo may have unintended consequences. The following will be classified as `internal` and I'm not sure they should be:

```
post /api/monitoring/v1/elasticsearch_settings/check/internal_monitoring
post /api/monitoring/v1/setup/collection/{clusterUuid}/disable_internal_collection
```

imo matching more strictly would reduce the potential for surprising outcomes which is better than missing a few routes. If routes should be internal they can add the `access: internal` flag. It looks like 99.9% of "internal" routes follow the convention of starting with `/internal/` so I would match that. Just my opinion, let me know what you think!

Reply:
you're totally right! I'm fixing it